src/components/DataTable.js

```diff
@@ -55,13 +57,20 @@ export const DataTable = ({
     } = useTable(
         {
             data,
-            columns
+            columns,
+            initialState: tableState
         },
         useFilters,
         useGlobalFilter,
         useSortBy
     );
 
+    useEffect(() => {
+        if (typeof setTableState == 'function') {
+            setTableState(state);
+        }
+    });
+
```

Comment:
Prettier/ESLint will resolve `==` to `===`; might refactor to circuit breaker arrow func.